File: math_stuff/less_interesting_languages/triangle.rb
# run with `ruby triangle.rb`

def triangle_number(n)
  (1..n).sum
end

def pyramid_number(n)
  (1..n).map{|n| n ** 2}.sum
end

puts "triangle numbers"
p (0..10).map{|n| triangle_number n}
puts "pyramid numbers"
p (0..10).map{|n| pyramid_number n}

File: math_stuff/less_interesting_languages/triangle.rs
// run with `rustc triangle.rs && ./triangle`

fn triangle(num: u32) -> u32 {
    (1..=num).sum()
}

fn pyramid(num: u32) -> u32 {
    (1..=num).map(|x| x*x).sum()
}

fn main() {
    println!("triangle numbers:");
    for i in 0..=10 {
        print!("{} ", triangle(i));
    }

    println!("");
    println!("pyramid numbers:");
    for i in 0..=10 {
        print!("{} ", pyramid(i));
    }
    println!("");
}

